checks.d/zk.py

```diff
@@ -96,6 +156,50 @@ def check(self, instance):
                 self.service_check('zookeeper.mode', status, message=message,
                                    tags=sc_tags)
 
+        if zk_version and LooseVersion(zk_version) > LooseVersion("3.4.0"):
+            try:
+                mntr_out = self._send_command('mntr', *cx_args)
+            except ZKConnectionFailure:
+                self.increment('zookeeper.timeouts')
+                self.report_instance_mode(hostname, 'down', tags)
+                raise
+            except:
+                e = sys.exc_info()[1]
+                print >> sys.stderr, "Error: %s" % e
+                self.increment('zookeeper.datadog_client_exception')
+                self.report_instance_mode(hostname, 'unknown', tags)
+                raise
+            else:
+                metrics, mode = self.parse_mntr(mntr_out)
+                mode_tag = "mode:%s" % mode
+                if mode != 'inactive':
+                    for name in metrics:
+                        self.gauge(name, metrics[name], tags=tags + [mode_tag])
```

Comment:
This is more of a style nitpick then a Python specific problem, but this code is adding 20 lines to a method that's already 80 lines long. It looks like this could be pulled out into it's own method